File: app/src/main/java/com/kp/bright/whatsapptickers/stickersmanage/StickerPackUtils.java
package com.kp.bright.whatsapptickers.stickersmanage;

import android.content.Context;
import android.util.Log;

import com.google.gson.Gson;
import com.kp.bright.whatsapptickers.whatsappsticker.StickerPackMetadata;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class StickerPackUtils {

    public static String TAG = "KNS_STICKER-->";

    public static void saveStickerPackMetadata(Context context, StickerPackMetadata metadata) {
        // Create JSON structure
        List<StickerPackMetadata> stickerPacks = new ArrayList<>();
        stickerPacks.add( metadata );
//        StickerPackJson stickerPackJson = new StickerPackJson();
//        stickerPackJson.sticker_packs = stickerPacks;

        // Convert to JSON
        Gson gson = new Gson();
        String json = gson.toJson( stickerPacks );

        // Create the stickers directory if it doesn't exist
        File stickersDir = new File( context.getExternalFilesDir(null), "stickers" );
        if (!stickersDir.exists()) {
            boolean dirCreated = stickersDir.mkdirs();
            if (!dirCreated) {
                Log.w( TAG, "Failed to create stickers directory: " + stickersDir.getAbsolutePath() );
                return;
            }
        }

        // Save JSON to file
        File file = new File( stickersDir, "sticker_packs.json" );
        Log.w( TAG, "saveStickerPackMetadata: " + file.getAbsolutePath() );
        try (FileWriter writer = new FileWriter( file )) {
            writer.write( json );
        } catch (IOException e) {
            Log.w( TAG, "IOException: " + file.getAbsolutePath() );
            e.printStackTrace();
        }
    }


    // Class representing the structure of the JSON file
    private static class StickerPackJson {
        List<StickerPackMetadata> sticker_packs;
    }
}

File: app/src/main/java/com/kp/bright/whatsapptickers/stickersmanage/StickerPackUtils.kt
package com.kp.bright.whatsapptickers.stickersmanage

import android.content.Context
import android.net.Uri
import android.util.Log
import com.google.gson.Gson
import com.kp.bright.whatsapptickers.whatsappsticker.StickerContentProvider
import com.kp.bright.whatsapptickers.whatsappsticker.StickerPackMetadata
import com.kp.bright.whatsapptickers.whatsappsticker.loadAllStickerPacksVisJson
import java.io.File
import java.io.FileWriter
import java.io.IOException

object StickerPackUtils {
    @JvmField
    var TAG: String = "KNS_STICKER-->"

    fun getMetadataURI(identifier: String?): Uri {
        return StickerContentProvider.AUTHORITY_URI.appendPath(StickerContentProvider.METADATA)
            .build()
    }

    fun getStickersURI(identifier: String?): Uri {
        return StickerContentProvider.AUTHORITY_URI.appendPath(StickerContentProvider.STICKERS)
            .appendPath(identifier).build()
    }

    fun getStickers_AssetsURI(identifier: String?, stickerName: String?): Uri {
        return StickerContentProvider.AUTHORITY_URI.appendPath(StickerContentProvider.STICKERS_ASSET)
            .appendPath(identifier).appendPath(stickerName).build()
    }

    fun saveStickerPackMetadata(context: Context, metadata: StickerPackMetadata) {
        // Create JSON structure
        val stickerPacks = loadAllStickerPacksVisJson(context)
        var s = stickerPacks.find { it.identifier == metadata.identifier }

        if (s != null) {
            return
        }
        stickerPacks.add(metadata)

        //        StickerPackJson stickerPackJson = new StickerPackJson();
//        stickerPackJson.sticker_packs = stickerPacks;

        // Convert to JSON
        val gson = Gson()
        val json = gson.toJson(stickerPacks)

        // Create the stickers directory if it doesn't exist
        val stickersDir = File(context.getExternalFilesDir(null), "stickers")

        if (!stickersDir.exists()) {
            val dirCreated = stickersDir.mkdirs()
            if (!dirCreated) {
                Log.w(TAG, "Failed to create stickers directory: " + stickersDir.absolutePath)
                return
            }
        }

        // Save JSON to file
        val file = File(stickersDir, "sticker_packs.json")
        Log.w(TAG, "saveStickerPackMetadata: " + file.absolutePath)
        try {
            FileWriter(file).use { writer ->
                writer.write(json)
            }
            Log.wtf(TAG, "saveStickerPackMetadata: $json")
        } catch (e: IOException) {
            Log.w(TAG, "IOException: " + file.absolutePath)
            e.printStackTrace()
        }
    }


    // Class representing the structure of the JSON file
    private class StickerPackJson {
        var sticker_packs: List<StickerPackMetadata>? = null
    }
}
